lib/src/main/java/com/alphawallet/token/tools/TokenDefinition.java

```diff
@@ -893,40 +893,22 @@ public void parseField(BigInteger tokenId, NonFungibleToken token) {
         }
     }
 
-    /**
-     * Check for 'appearance' attribute set
-     * @param tag
-     * @return
-     */
-    public String getAppearance(String tag)
-    {
-        Map<String, String> appearanceSet = attributeSets.get("appearance");
-        if (appearanceSet != null)
-        {
-            return appearanceSet.get(tag);
-        }
-        else
-        {
-            return "";
-        }
-    }
-
     /**
      * Check for 'cards' attribute set
      * @param tag
      * @return
      */
     public String getCardData(String tag)
     {
+        String view = null;
         Map<String, String> appearanceSet = attributeSets.get("cards");
         if (appearanceSet != null)
         {
-            return appearanceSet.get(tag);
-        }
-        else
-        {
-            return "";
+            view = appearanceSet.get(tag);
+            if (view == null && tag.equals("item-view")) view = getCardData("view-iconified"); //deal with legacy views
```

Comment:
@JamesSmartCell we can probably just drop the old way, best to leave behind all the legacy stuff and move on

Reply:
It's may not be a good idea to just drop something like that - no language I know ever phased out anything abruptly - you mark it deprecated then remove a few months later. Depends how many people wrote scripts already. If they couldn't get them working because of this ...

Reply:
Here's how it will work: let's remove the legacy handling but add an error message to show what happened should anyone ever come across this.

Reply:
@JamesSmartCell I think this point is moot because we don't support legacy versions anyway and the new schema will not validate if you use the old way; therefore supporting the old syntax doesn't add anything. Regarding phasing out, swift does this all the time, just ask @hboon. 

Also remember that there are few people writing them right now so we might as well eliminate the older versions while we still can.

What do you think?

Reply:
@colourful-land

Reply:
@JamesSmartCell made the point that for future proofing we should add a warning that it is old, this is a good idea but a different PR.

Reply:
@James-Sangalli Removed legacy support, added error feedback as per suggestion from @colourful-land to show the view is deprecated and link to schema description.

Reply:
Old script will still work because old script provides its own wrapper (adding one doesn't change anything) and the reactJS update call is designed (by @hboon) to be backward compatible. That's why I asked this to be done first, before even forking upgrade-schema.